File: 0109. 有序链表转换二叉搜索树/递归.cpp
// 44ms
class Solution {
public:
    TreeNode* sortedListToBST(ListNode* head) {
        if (head == nullptr) {
            return nullptr;
        } else {
            int len = 0;
            ListNode* t = head;
            while (t != nullptr) {
                len += 1;
                t = t->next;
            }

            if (len == 1) {
                return new TreeNode(head->val);
            } else {
                int stop = len / 2 - 1;
                int count = 0;
                ListNode* pre = head;
                while (count != stop) {
                    count += 1;
                    pre = pre->next;
                }
                ListNode* mid = pre->next;
                pre->next = nullptr;
                TreeNode* root = new TreeNode(mid->val);
                root->left = sortedListToBST(head);
                root->right = sortedListToBST(mid->next);
                return root;
            }
        }
    }
};

File: 0701. 二叉搜索树中的插入操作/递归.js
// 144ms
var insertIntoBST = function (root, val) {
    var current = root, node = new TreeNode(val)
    if (root != null) {
        while (true) {
            if (val < current.val) {
                if (current.left == null) {
                    current.left = node
                    break
                } else {
                    current = current.left
                }
            } else {
                if (current.right == null) {
                    current.right = node
                    break
                } else {
                    current = current.right
                }
            }
        }
        return root
    } else {
        return node
    }
};